pkg/core/dao/cacheddao.go

```diff
@@ -200,5 +211,111 @@ func (cd *Cached) GetWrapped() DAO {
 		cd.unspents,
 		cd.balances,
 		cd.transfers,
+		cd.storage,
```

Comment:
So wrapped instances are actually sharing storage?

Reply:
Yes this is exactly what we need to save cache between tx executions in the same block.

Reply:
But then some changes from failed transactions might leak into the lower level cache and not be discarded?

Reply:
Actually not, because flushing is done only on success. But we need this so that even which were read in failed transactions are present in the cache.

Reply:
> even items which were read in failed transactions are present in the cache.

Yeah, sounds like the path to success!..

Reply:
But then you're also caching failed writes, aren't you?

Reply:
No, because `get` goes down->up, and `put`/`del` goes up->down, thus is present only in the upper cache.

Reply:
This is what happens during tx execution: Wrapped(D) -> Wrapped(CD1) -> CD2
Items which were in `get` were requested from D, then from CD1 (and stored in cache), then from CD2 (and stored in cache). Items which were in `put`/`del` are only in CD2.
If persist is not called, CD2 storage is discarded is not flushed to CD1.

Reply:
OK, the trick is that `GetWrapped` is actually an internal `dao` interface and it not used outside of it (where `NewCached` is being used and that creates a new proper `storage`.